jetpack-start/wp-cli-api.php

```diff
@@ -61,34 +66,64 @@ public function connect( $args, $assoc_args ) {
 			WP_CLI::warning( 'JETPACK_DEV_DEBUG mode is enabled. Please remove the constant after connection.' );
 		}
 
-		$force_connection = WP_CLI\Utils\get_flag_value( $assoc_args, 'force', false );
-		if ( ! $force_connection && Jetpack::is_active() ) {
-			WP_CLI::error( 'Jetpack is already active; bailing. Run this command with `--force` to bypass this check or disconnect Jetpack before continuing.' );
-		}
+		$sites = [ false ];
 
-		WP_CLI::line( '-- Verifying VIP machine user exists (or creating one, if not)' );
-		$user = $this->maybe_create_user();
-		if ( is_wp_error( $user ) ) {
-			WP_CLI::error( $user->get_error_message() );
+		$network = WP_CLI\Utils\get_flag_value( $assoc_args, 'network', false );
+		if ( $network && is_multisite() ) {
+			$sites = get_sites( [
+				'public'   => null,
+				'archived' => 0,
+				'spam'     => 0,
+				'deleted'  => 0,
+				'fields'   => 'ids',
+			] );
 		}
 
-		WP_CLI::line( '-- Fetching keys from Jetpack Start API' );
-		$data = $this->api_fetch_keys( $user );
-		if ( is_wp_error( $data ) ) {
-			$message = $data->get_error_message();
-			// Use strpos because our API method appends stuff to the error message
-			if ( false !== strpos( $message, self::API_ERROR_EXISTING_SUBSCRIPTION ) ) {
-				$message = 'There is an existing Jetpack Start subcription for this site. Please disconnect using the `cancel` subcommand and try again.';
-			} elseif ( false !== strpos( $message, self::API_ERROR_USER_PERMISSIONS ) ) {
-				$message = sprintf( 'This site already has an existing Jetpack shadow site but the `%s` WP.com user is either not a member of the site or not an administrator. Please use `add_user_to_blog` on your WP.com sandbox to add the account before continuing.', WPCOM_VIP_MACHINE_USER_LOGIN );
+		foreach ( $sites as $site ) {
+			// Network activation request
+			if ( is_int( $site ) ) {
+				switch_to_blog( $site );
+
+				WP_CLI::line( sprintf( 'Starting %s (site %d)', home_url( '/' ), $site ) );
+			}
+
+			$force_connection = WP_CLI\Utils\get_flag_value( $assoc_args, 'force', false );
+			if ( ! $force_connection && Jetpack::is_active() ) {
+				WP_CLI::error( 'Jetpack is already active; bailing. Run this command with `--force` to bypass this check or disconnect Jetpack before continuing.' );
 			}
-			WP_CLI::error( 'Failed to fetch keys from Jetpack Start: ' . $message );
-		}
 
-		WP_CLI::line( '-- Adding keys to site' );
-		$this->install_keys( $data );
+			WP_CLI::line( '-- Verifying VIP machine user exists (or creating one, if not)' );
```

Comment:
I think if we split off the connection logic into its own method, that would:

- simplify the multisite vs single site handling here (no more hacky `is_int` check)
- make it easier to handle the `WP_CLI::error` cases by returning early from the method.